src/proc_interrupts.c

```diff
@@ -154,19 +172,20 @@ int do_proc_interrupts(int update_every, unsigned long long dt) {
                 char title[100+1];
                 snprintfz(title, 100, "CPU%d Interrupts", c);
                 st = rrdset_create("cpu", id, NULL, "interrupts", "cpu.interrupts", title, "interrupts/s", 1100 + c, update_every, RRDSET_TYPE_STACKED);
-
-                for(l = 0; l < lines ;l++) {
-                    struct interrupt *irr = irrindex(irrs, l, cpus);
-                    if(unlikely(!irr->used)) continue;
-                    rrddim_add(st, irr->id, irr->name, 1, 1, RRDDIM_INCREMENTAL);
-                }
             }
             else rrdset_next(st);
 
             for(l = 0; l < lines ;l++) {
                 struct interrupt *irr = irrindex(irrs, l, cpus);
                 if(unlikely(!irr->used)) continue;
-                rrddim_set(st, irr->id, irr->value[c]);
+                if(unlikely(!irr->cpu[c].rd || strncmp(irr->name, irr->cpu[c].rd->name, MAX_INTERRUPT_NAME != 0))) {
```

Comment:
Right, and these ones are unneeded, we can just reset the main `RRDDIM` and every cpu one in the previous check. I will submit a new patch with this.